src/fileio.cpp

```diff
@@ -977,55 +973,63 @@ bool DoScanWorkingDirectory()
 	return _searchpaths[SP_PERSONAL_DIR] != tmp;
 }
 
+/**
+ * Gets the home directory of the user.
+ * May return an empty string in the unlikely scenario that the home directory cannot be found.
+ * @return User's home directory
+ */
+static std::string GetHomeDir()
+{
+#ifdef __HAIKU__
+	BPath path;
+	find_directory(B_USER_SETTINGS_DIRECTORY, &path);
+	return std::string(path.Path());
+#else
+	const char *home_env = getenv("HOME"); // Stack var, shouldn't be freed
+	if (home_env != nullptr) {
+		return std::string(home_env);
+	} else {
```

Comment:
doing an else after a return always feels weird. Is there a benefit of doing this? Leaving it out sounds more readable to me, as it becomes a chain of events:

- try env
- try getpwuid
?